src/rigid_pendulum_poc/solver.h

```diff
@@ -1,10 +1,17 @@
 #pragma once
 
-#include<Kokkos_Core.hpp>
+#include <Kokkos_Core.hpp>
 
 namespace openturbine::rigid_pendulum {
 
-[[nodiscard]] int solve_linear_system(Kokkos::View<double**> system, Kokkos::View<double*> solution);
+/// @brief Solve a linear system of equations using LAPACKE's dgesv
+/// @details This function solves a linear system of equations using LAPACKE's
+///     dgesv function. The system is of the form Ax = b, where A is a square matrix
+///     (nxn) of coefficients, x is a vector (nx1) of unknowns, and b is a vector
+///     (nx1) of right-hand side values. The solution is stored in the vector b.
+/// @param system A matrix of coefficients
+/// @param solution A vector of right-hand side values
+/// @return An integer indicating the success of the solution
+int solve_linear_system(Kokkos::View<double**>, Kokkos::View<double*>);
```

Comment:
Do we want to keep nodiscard?  It is a not so gentle reminder for the caller to check the error value.  Or we could check the lapack error value and throw inside the function if it is anything other than 0.

Reply:
Sounds good to me